tests/php/ORM/Filters/ExactMatchFilterTest.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace SilverStripe\ORM\Tests\Filters;
+
+use SilverStripe\Core\Config\Config;
+use SilverStripe\Dev\SapphireTest;
+use SilverStripe\ORM\Filters\ExactMatchFilter;
+use SilverStripe\ORM\Tests\Filters\ExactMatchFilterTest\Task;
+use SilverStripe\ORM\Tests\Filters\ExactMatchFilterTest\Project;
+
+class ExactMatchFilterTest extends SapphireTest
+{
+    protected static $fixture_file = 'ExactMatchFilterTest.yml';
+
+    protected static $extra_dataobjects = [
+        Task::class,
+        Project::class,
+    ];
+
+    /**
+     * @dataProvider provideUsePlaceholders
+     */
+    public function testUsePlaceholders(?bool $expectedID, ?bool $expectedTitle, bool $config, callable $fn): void
+    {
+        Config::modify()->set(ExactMatchFilter::class, 'use_placeholders_for_integer_ids', $config);
+        [$idQueryUsesPlaceholders, $titleQueryUsesPlaceholders] = $this->usesPlaceholders($fn);
+        $this->assertSame($expectedID, $idQueryUsesPlaceholders);
+        $this->assertSame($expectedTitle, $titleQueryUsesPlaceholders);
+    }
+
+    public function provideUsePlaceholders(): array
+    {
+        $ids = [1, 2, 3];
+        $taskTitles = array_map(fn($i) => "Task $i", $ids);
+        return [
+            'primary key' => [
+                'expectedID' => false,
+                'expectedTitle' => null,
+                'config' => false,
+                'fn' => fn() => Task::get()->byIDs($ids)
+            ],
+            'foriegn key' => [
```

Comment:
```suggestion
            'primary key on relation' => [
                'expectedID' => false,
                'expectedTitle' => null,
                'config' => false,
                'fn' => fn() => Project::get()->filter('Tasks.ID')
            ],
            'foriegn key' => [
```
Just to make sure ID fields _on relations_ (i.e. using dot syntax) are also covered.

Reply:
Updated